src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/PackageManagerUICommandHandler.cs

```diff
@@ -328,72 +329,55 @@ private async Task<System.Windows.Window> CreateNewWindowFrameAsync(string proje
-            // var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(vsShellInterop.SVsUIShell)) as vsShellInterop.IVsUIShell;
-            // Assumes.Present(uiShell);
-            // var ppunkToolView = IntPtr.Zero;
-            // var hr = 0;
-            // int[] pfDefaultPosition = null;
-
-            // try
-            // {
-            //     ppunkToolView = Marshal.GetIUnknownForObject(windowPane);
-
-            //     hr = uiShell.CreateToolWindow(
-            //         windowFlags,
-            //         0, //dwToolWindowId - singleInstance = 0
-            //         ppunkToolView, // punkTool
-            //         Guid.Empty,  //rclsidTool = GUID_NULL
-            //         Guid.NewGuid(), // TODO: should be projectGuid...so persistance info works.
-            //         Guid.Empty,  //reserved - do not use - GUID_NULL
-            //         null, // IServiceProvider
-            //         caption,
-            //         pfDefaultPosition,
-            //         out windowFrame);
-
-            //     // hr = uiShell.CreateDocumentWindow(
-            //     //     windowFlags,
-            //     //     documentName,
-            //     //     (vsShellInterop.IVsUIHierarchy)hier,
-            //     //     itemId,
-            //     //     ppunkDocView,
-            //     //     ppunkDocData,
-            //     //     ref guidEditorType,
-            //     //     null,
-            //     //     ref guidCommandUI,
-            //     //     null,
-            //     //     caption,
-            //     //     string.Empty,
-            //     //     null,
-            //     //     out windowFrame);
-            //     if (windowFrame != null)
-            //     {
-            //         WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmUI);
-            //         WindowFrameHelper.DisableWindowAutoReopen(windowFrame);
-            //     }
-            // }
-            // finally
-            // {
-            //     if (ppunkToolView != IntPtr.Zero)
-            //     {
-            //         Marshal.Release(ppunkDocData);
-            //     }
-
-            //     if (windowPane != null)
-            //     {
-            //         windowPane.Dispose();
-            //     }
-            // }
+            //var control = new Button() { Content = "Package Manager UI - under construction" };
+            //var win = new System.Windows.Window();
+            //win.Title = caption;
+            //win.Content = control;
+
+            vsShellInterop.IVsWindowFrame windowFrame;
+            var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(vsShellInterop.SVsUIShell)) as vsShellInterop.IVsUIShell;
+            Assumes.Present(uiShell);
+            var hr = 0;
+            int[] pfDefaultPosition = null;
+
+            try
+            {
+                uint createFlag = (uint)__VSCREATETOOLWIN.CTW_fInitNew;
+                hr = uiShell.CreateToolWindow(
```

Comment:
can you create a follow up issue, to move to AsyncPackage apis for tool window, per that email advice.